query/src/test/java/org/infinispan/query/tx/NonLocalIndexingTest.java

```diff
@@ -61,7 +59,7 @@ public void testQueryAfterAddingNewNode() throws Exception {
       store("Astronaut", new AnotherGrassEater("Astronaut", "is having a hard time to find grass"), cache(1));
       //replacing same key with a new type:
       assertFind("cat", 0);
-      assertFind("grass", 1);
+      assertFind("grass", 0);
```

Comment:
I'm confused by this. Why 0?

Reply:
The API that worked on Lucene queries would do a query targeting all entities in the cache (Person and AnotherGrassEater), with my changes it only targets the Person entity